### PR TITLE
feat: Make new endpoint to return most used locations

### LINKED ISSUE #5945: New Endpoint for Most used Event-Locations
Server issue of [fossasia/open-event-frontend#2970](https://github.com/fossasia/open-event-frontend/issues/2970)

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5945 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
New endpoint to return most used locations

### CODE REVIEW

**Comment by prateekj117** [comment]
@iamareebjamal Instead of returning a result, the query here is returning the sql statement. Any idea why?

**Comment by iamareebjamal** [comment]
Please add `wip:` in the title for work in progress PRs

**Comment by uds5501** [comment]
> @iamareebjamal Instead of returning a result, the query here is returning the sql statement. Any idea why?

You are forgetting a `.one()` or `.all()` (use as you require)

**Comment by prateekj117** [comment]
@iamareebjamal Please let me know what other things are required in this PR.

**Comment by iamareebjamal** [comment]
Won't review WIP PR

File: app/api/events.py [hunk 1]
```diff
@@ -575,3 +579,9 @@ def clear_export_urls(event):
     event.xcal_url = None
     event.pentabarf_url = None
     save_to_db(event)
+
+
+@event_location_routes.route('/most-used-locations', methods=['GET'])
+def most_used_location():
+    locations = db.session.query(Event.location_name).group_by(Event.location_name).order_by(desc(func.count(Event.location_name))).all()
```
[the remark this answers is not in the document]
Comment (uds5501) [reply app/api/events.py]: @prateekj117 Please fix the hound error.

File: app/api/events.py [hunk 2]
```diff
@@ -575,3 +579,9 @@ def clear_export_urls(event):
...
+    return jsonify({"locations": locations})
```
Comment (uds5501) [inline app/api/events.py]: @prateekj117 I am a bit skeptical about this one. can you please attach a sample response on this endpoint? meanwhile I will clone it and check on my local. Thanks!

Comment (prateekj117) [reply app/api/events.py]: @uds5501 

> {
    "locations": [
        {
            "location_name": "Noida"
        },
        {
            "location_name": "ABC"
        },
        {
            "location_name": "Delhi"
        },
        {
            "location_name": "example1"
        }
    ]
}

**Comment by iamareebjamal** [comment]
Follow JSONAPI spec response and endpoint style please

**Comment by iamareebjamal** [comment]
Still incorrect

**Comment by iamareebjamal** [comment]
Still incorrect

File: app/api/events.py [hunk 3]
```diff
@@ -575,3 +580,10 @@ def clear_export_urls(event):
     event.xcal_url = None
     event.pentabarf_url = None
     save_to_db(event)
+
+
+@event_location_routes.route('/most-used-locations', methods=['GET'])
+def most_used_location():
+    locations = db.session.query(Event.location_name).group_by(Event.location_name). \
```
Comment (Anupam-dagar) [inline app/api/events.py]: For line continuation, you should try avoiding back slashes, instead break on parenthesis. Something like this
```suggestion
    locations = db.session.query(Event.location_name).group_by(
    Event.location_name).
```

Comment (mrsaicharan1) [reply app/api/events.py]: Yeah, don't need backslashes when parenthesis are present.

**Comment by prateekj117** [comment]
@iamareebjamal Should we close it now?

**Comment by iamareebjamal** [comment]
Discuss this in next meeting with peers what is to be done

**Pull request closed without merging**